strawberryfields/circuitspecs/chip0.py

```diff
@@ -51,7 +59,130 @@ class Chip0Specs(CircuitSpecs):
         Rgate({internal_phase_0}) | [2]
         BSgate(pi/4, pi/2) | [2, 3]
 
+        # final local phases
+        Rgate({phi0}) | 0
```

Comment:
Might be good for @lneuhaus to verify this modified circuit specification

Reply:
Unitary decomposition always produces a diagonal array of local phase shifts. These can go anywhere, as they can be commuted through the beamsplitter arrays.

The `rectangular` function sandwiches them between two beamsplitter arrays, whereas `rectangular_symmetric` and `rectangular_phase_end` commute them to the end of the beamsplitter array --- this is preferred in experimental linear optics (https://github.com/XanaduAI/strawberryfields/pull/44).

So they are unavoidable, but having them at the end means they shouldn't affect the photon number statistics, and they can probably be ignored in hardware (@lneuhaus, did I get this right?)

Reply:
Josh and I concluded this was the best way to modify the circuit spec to map an arbitrary unitary to the chip topology: if we do Fock measurements, this final phase is just a delay line before the photon number resolving detector, so it does not affect the measurement outcome. And in case one would perform Homodyne measurements one day, this final phase can be used comfortably to tweak the relative phase w.r.t. local oscillator (whose initial phase offset will probably be random). 

Only two minor comments:
- as discussed with Josh, another option could be to absorb an arbitrary phase shift into the FockMeasurement gate. This is just a suggestion based on the observation that any phase shift before a Fock Measurement does not affect the measurement outcome, and if it makes you feel uncomfortable, do not do it. 
- to keep names in line with `internal_phase_x` and `external_phase_y`, I suggest to call these final phases `final_phase_z` as this is more explicit than `phi_z`.

Reply:
> to keep names in line with internal_phase_x and external_phase_y, I suggest to call these final phases final_phase_z as this is more explicit than phi_z.

done :+1: